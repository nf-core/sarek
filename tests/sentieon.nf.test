nextflow_pipeline {

    name "Test sentieon"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_sarek"
    tag "sentieon"
    tag "cpu"

    test("-profile test --aligner sentieon-bwamem --tools null -stub") {

        options "-stub"

        when {
            params {
                outdir = "$outputDir"
                aligner = "sentieon-bwamem"
                tools = ''
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: All bam files
            def bam_files  = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            // cram_files: All cram files
            def cram_files  = getAllFilesFromDir(params.outdir, include: ['**/*.cram'])
            // Fasta file for cram verification with nft-bam
            def fasta_base  = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            def fasta       = fasta_base + 'genomics/homo_sapiens/genome/genome.fasta'
            // vcf_files: All vcf files
            def vcf_files  = getAllFilesFromDir(params.outdir, include: ['**/*.vcf{,.gz}'])
            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow and pipeline versions are removed (all from the workflow key)
                    removeFromYamlMap("${outputDir}/pipeline_info/nf_core_sarek_software_mqc_versions.yml", "Workflow"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path.isEmpty() ? 'No stable content' : stable_path,
                    // All cram files
                    bam_files.isEmpty() ? 'No BAM files' : bam_files.collect { file -> file.getName() + ":md5," +  bam(file.toString()).readsMD5 },
                    // All cram files
                    cram_files.isEmpty() ? 'No CRAM files' : cram_files.collect { file -> file.getName() + ":md5," +  cram(file.toString(), fasta).readsMD5 },
                    // All vcf files
                    vcf_files.isEmpty() ? 'No VCF files' : vcf_files.collect { file -> file.getName() + ":md5," + path(file.toString()).vcf.variantsMD5 }
                ).match() }
            )
        }
    }
}
