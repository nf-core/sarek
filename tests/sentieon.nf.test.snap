{
    "stub": {
        "content": [
            23,
            {
                "BCFTOOLS_STATS": {
                    "bcftools": 1.2
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GATK4_APPLYBQSR": {
                    "gatk4": "4.5.0.0"
                },
                "GATK4_BASERECALIBRATOR": {
                    "gatk4": "4.5.0.0"
                },
                "GATK4_MARKDUPLICATES": {
                    "gatk4": "4.5.0.0",
                    "samtools": "1.19.2"
                },
                "INDEX_CRAM": {
                    "samtools": 1.21
                },
                "MOSDEPTH": {
                    "mosdepth": "0.3.8"
                },
                "SAMTOOLS_STATS": {
                    "samtools": 1.21
                },
                "SENTIEON_BWAMEM": {
                    "sentieon": 202308.03,
                    "bwa": "0.7.17-r1188"
                },
                "STRELKA_SINGLE": {
                    "strelka": "2.9.10"
                },
                "VCFTOOLS_TSTV_COUNT": {
                    "vcftools": "0.1.16"
                },
                "Workflow": {
                    "nf-core/sarek": "v3.6.0dev"
                }
            },
            [
                "csv",
                "csv/markduplicates.csv",
                "csv/markduplicates_no_table.csv",
                "csv/recalibrated.csv",
                "csv/variantcalled.csv",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_sarek_software_mqc_versions.yml",
                "preprocessing",
                "preprocessing/markduplicates",
                "preprocessing/markduplicates/test",
                "preprocessing/markduplicates/test/test.cram",
                "preprocessing/markduplicates/test/test.cram.crai",
                "preprocessing/recal_table",
                "preprocessing/recal_table/test",
                "preprocessing/recal_table/test/test.recal.table",
                "preprocessing/recalibrated",
                "preprocessing/recalibrated/test",
                "preprocessing/recalibrated/test/test.recal.cram",
                "preprocessing/recalibrated/test/test.recal.cram.crai",
                "reference",
                "reports",
                "reports/bcftools",
                "reports/bcftools/strelka",
                "reports/bcftools/strelka/test",
                "reports/bcftools/strelka/test/test.strelka.variants.bcftools_stats.txt",
                "reports/fastqc",
                "reports/fastqc/test-test_L1",
                "reports/fastqc/test-test_L1/test-test_L1.html",
                "reports/fastqc/test-test_L1/test-test_L1.zip",
                "reports/fastqc/test-test_L2",
                "reports/fastqc/test-test_L2/test-test_L2.html",
                "reports/fastqc/test-test_L2/test-test_L2.zip",
                "reports/markduplicates",
                "reports/markduplicates/test",
                "reports/markduplicates/test/test.md.cram.metrics",
                "reports/mosdepth",
                "reports/mosdepth/test",
                "reports/mosdepth/test/test.md.global.dist.txt",
                "reports/mosdepth/test/test.md.per-base.bed.gz",
                "reports/mosdepth/test/test.md.per-base.bed.gz.csi",
                "reports/mosdepth/test/test.md.per-base.d4",
                "reports/mosdepth/test/test.md.quantized.bed.gz",
                "reports/mosdepth/test/test.md.quantized.bed.gz.csi",
                "reports/mosdepth/test/test.md.region.dist.txt",
                "reports/mosdepth/test/test.md.regions.bed.gz",
                "reports/mosdepth/test/test.md.regions.bed.gz.csi",
                "reports/mosdepth/test/test.md.summary.txt",
                "reports/mosdepth/test/test.md.thresholds.bed.gz",
                "reports/mosdepth/test/test.md.thresholds.bed.gz.csi",
                "reports/mosdepth/test/test.recal.global.dist.txt",
                "reports/mosdepth/test/test.recal.per-base.bed.gz",
                "reports/mosdepth/test/test.recal.per-base.bed.gz.csi",
                "reports/mosdepth/test/test.recal.per-base.d4",
                "reports/mosdepth/test/test.recal.quantized.bed.gz",
                "reports/mosdepth/test/test.recal.quantized.bed.gz.csi",
                "reports/mosdepth/test/test.recal.region.dist.txt",
                "reports/mosdepth/test/test.recal.regions.bed.gz",
                "reports/mosdepth/test/test.recal.regions.bed.gz.csi",
                "reports/mosdepth/test/test.recal.summary.txt",
                "reports/mosdepth/test/test.recal.thresholds.bed.gz",
                "reports/mosdepth/test/test.recal.thresholds.bed.gz.csi",
                "reports/samtools",
                "reports/samtools/test",
                "reports/samtools/test/test.md.cram.stats",
                "reports/samtools/test/test.recal.cram.stats",
                "reports/vcftools",
                "reports/vcftools/strelka",
                "reports/vcftools/strelka/test",
                "reports/vcftools/strelka/test/test.strelka.variants.012",
                "reports/vcftools/strelka/test/test.strelka.variants.012.indv",
                "reports/vcftools/strelka/test/test.strelka.variants.012.pos",
                "reports/vcftools/strelka/test/test.strelka.variants.BEAGLE.GL",
                "reports/vcftools/strelka/test/test.strelka.variants.BEAGLE.PL",
                "reports/vcftools/strelka/test/test.strelka.variants.FILTER.summary",
                "reports/vcftools/strelka/test/test.strelka.variants.FORMAT",
                "reports/vcftools/strelka/test/test.strelka.variants.INFO",
                "reports/vcftools/strelka/test/test.strelka.variants.LROH",
                "reports/vcftools/strelka/test/test.strelka.variants.Tajima.D",
                "reports/vcftools/strelka/test/test.strelka.variants.TsTv",
                "reports/vcftools/strelka/test/test.strelka.variants.TsTv.count",
                "reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual",
                "reports/vcftools/strelka/test/test.strelka.variants.TsTv.summary",
                "reports/vcftools/strelka/test/test.strelka.variants.bcf",
                "reports/vcftools/strelka/test/test.strelka.variants.diff.discordance.matrix",
                "reports/vcftools/strelka/test/test.strelka.variants.diff.indv",
                "reports/vcftools/strelka/test/test.strelka.variants.diff.indv_in_files",
                "reports/vcftools/strelka/test/test.strelka.variants.diff.sites",
                "reports/vcftools/strelka/test/test.strelka.variants.diff.sites_in_files",
                "reports/vcftools/strelka/test/test.strelka.variants.diff.switch",
                "reports/vcftools/strelka/test/test.strelka.variants.frq",
                "reports/vcftools/strelka/test/test.strelka.variants.frq.count",
                "reports/vcftools/strelka/test/test.strelka.variants.gdepth",
                "reports/vcftools/strelka/test/test.strelka.variants.geno.chisq",
                "reports/vcftools/strelka/test/test.strelka.variants.geno.ld",
                "reports/vcftools/strelka/test/test.strelka.variants.hap.ld",
                "reports/vcftools/strelka/test/test.strelka.variants.hapcount",
                "reports/vcftools/strelka/test/test.strelka.variants.het",
                "reports/vcftools/strelka/test/test.strelka.variants.hwe",
                "reports/vcftools/strelka/test/test.strelka.variants.idepth",
                "reports/vcftools/strelka/test/test.strelka.variants.ifreqburden",
                "reports/vcftools/strelka/test/test.strelka.variants.imiss",
                "reports/vcftools/strelka/test/test.strelka.variants.impute.hap",
                "reports/vcftools/strelka/test/test.strelka.variants.impute.hap.indv",
                "reports/vcftools/strelka/test/test.strelka.variants.impute.hap.legend",
                "reports/vcftools/strelka/test/test.strelka.variants.indel.hist",
                "reports/vcftools/strelka/test/test.strelka.variants.interchrom.geno.ld",
                "reports/vcftools/strelka/test/test.strelka.variants.interchrom.hap.ld",
                "reports/vcftools/strelka/test/test.strelka.variants.kept.sites",
                "reports/vcftools/strelka/test/test.strelka.variants.ldepth",
                "reports/vcftools/strelka/test/test.strelka.variants.ldepth.mean",
                "reports/vcftools/strelka/test/test.strelka.variants.ldhat.locs",
                "reports/vcftools/strelka/test/test.strelka.variants.ldhat.sites",
                "reports/vcftools/strelka/test/test.strelka.variants.list.geno.ld",
                "reports/vcftools/strelka/test/test.strelka.variants.list.hap.ld",
                "reports/vcftools/strelka/test/test.strelka.variants.lmiss",
                "reports/vcftools/strelka/test/test.strelka.variants.lqual",
                "reports/vcftools/strelka/test/test.strelka.variants.map",
                "reports/vcftools/strelka/test/test.strelka.variants.mendel",
                "reports/vcftools/strelka/test/test.strelka.variants.ped",
                "reports/vcftools/strelka/test/test.strelka.variants.relatedness",
                "reports/vcftools/strelka/test/test.strelka.variants.relatedness2",
                "reports/vcftools/strelka/test/test.strelka.variants.removed.sites",
                "reports/vcftools/strelka/test/test.strelka.variants.singletons",
                "reports/vcftools/strelka/test/test.strelka.variants.sites.pi",
                "reports/vcftools/strelka/test/test.strelka.variants.snpden",
                "reports/vcftools/strelka/test/test.strelka.variants.tfam",
                "reports/vcftools/strelka/test/test.strelka.variants.tped",
                "reports/vcftools/strelka/test/test.strelka.variants.vcf",
                "reports/vcftools/strelka/test/test.strelka.variants.weir.fst",
                "reports/vcftools/strelka/test/test.strelka.variants.windowed.pi",
                "variant_calling",
                "variant_calling/strelka",
                "variant_calling/strelka/test",
                "variant_calling/strelka/test/test.strelka.genome.vcf.gz",
                "variant_calling/strelka/test/test.strelka.genome.vcf.gz.tbi",
                "variant_calling/strelka/test/test.strelka.variants.vcf.gz",
                "variant_calling/strelka/test/test.strelka.variants.vcf.gz.tbi"
            ],
            [
                "test.cram:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.cram.crai:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.bcftools_stats.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test-test_L1.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test-test_L1.zip:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test-test_L2.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test-test_L2.zip:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.global.dist.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.per-base.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.md.per-base.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.per-base.d4:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.quantized.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.md.quantized.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.region.dist.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.regions.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.md.regions.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.thresholds.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.md.thresholds.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.global.dist.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.per-base.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.recal.per-base.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.per-base.d4:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.quantized.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.recal.quantized.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.region.dist.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.regions.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.recal.regions.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.thresholds.bed.gz:md5,68b329da9893e34099c7d8ad5cb9c940",
                "test.recal.thresholds.bed.gz.csi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.md.cram.stats:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.recal.cram.stats:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.012:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.012.indv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.012.pos:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.BEAGLE.GL:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.BEAGLE.PL:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.FILTER.summary:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.FORMAT:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.INFO:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.LROH:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.Tajima.D:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.TsTv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.TsTv.count:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.TsTv.summary:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.bcf:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.diff.discordance.matrix:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.diff.indv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.diff.indv_in_files:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.diff.sites:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.diff.sites_in_files:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.diff.switch:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.frq:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.frq.count:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.gdepth:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.geno.chisq:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.geno.ld:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.hap.ld:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.hapcount:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.het:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.hwe:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.idepth:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.ifreqburden:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.imiss:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.impute.hap:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.impute.hap.indv:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.impute.hap.legend:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.indel.hist:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.interchrom.geno.ld:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.interchrom.hap.ld:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.kept.sites:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.ldepth:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.ldepth.mean:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.ldhat.locs:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.ldhat.sites:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.list.geno.ld:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.list.hap.ld:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.lmiss:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.lqual:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.map:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.mendel:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.ped:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.relatedness:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.relatedness2:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.removed.sites:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.singletons:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.sites.pi:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.snpden:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.tfam:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.tped:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.vcf:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.weir.fst:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test.strelka.variants.windowed.pi:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                [
                    "test.cram",
                    "d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "test.recal.cram",
                    "d41d8cd98f00b204e9800998ecf8427e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-09T10:15:55.105507"
    }
}
